pkg/virt-controller/watch/vmi.go

```diff
@@ -1424,12 +1508,35 @@ func (c *VMIController) createAttachmentPodTemplate(volume *virtv1.Volume, virtl
 		return nil, err
 	}
 	if populated {
-		pod, err := c.templateService.RenderHotplugAttachmentPodTemplate(volume, virtlauncherPod, vmi, pvc.Name, isBlock)
+		pod, err := c.templateService.RenderHotplugAttachmentPodTemplate(volume, virtlauncherPod, vmi, pvc.Name, isBlock, false)
 		return pod, err
 	}
 	return nil, nil
 }
 
+func (c *VMIController) createAttachmentPopulateTriggerPodTemplate(volume *virtv1.Volume, virtlauncherPod *k8sv1.Pod, vmi *virtv1.VirtualMachineInstance) (*k8sv1.Pod, error) {
+	var claimName string
+	if volume.DataVolume != nil {
+		// TODO, look up the correct PVC name based on the datavolume, right now they match, but that will not always be true.
+		claimName = volume.DataVolume.Name
+	} else if volume.PersistentVolumeClaim != nil {
+		claimName = volume.PersistentVolumeClaim.ClaimName
+	}
+	if claimName == "" {
+		return nil, errors.New("Unable to hotplug, claim not PVC or Datavolume")
+	}
+
+	pvc, exists, isBlock, err := kubevirttypes.IsPVCBlockFromClient(c.clientset, virtlauncherPod.Namespace, claimName)
```

Comment:
This needs to be using `IsPVCBlockFromStore` since we already have a pvc informer in the vmi controller. Make sure all instances of IsPVCBlockFromClient are transformed to IsPVCBlockFromStore in this controller.

Reply:
Done